tests/aggregation/test_filter_argument.py

```diff
@@ -192,3 +192,29 @@ def test_filtered_aggregate_on_exists(self):
             ),
         )
         self.assertEqual(aggregate, {"max_rating": 4.5})
+
+    def test_filtered_aggregate_empty_condition(self):
+        book = Book.objects.annotate(
+            authors_count=Count(
+                "authors",
+                filter=Q(authors__name__in=[]),
+            ),
+        ).get(pk=self.b1.pk)
+        self.assertEqual(book.authors_count, 0)
+        aggregate = Book.objects.aggregate(
+            max_rating=Max("rating", filter=Q(rating__in=[]))
+        )
+        self.assertEqual(aggregate, {"max_rating": None})
+
+    def test_filtered_aggregate_full_condition(self):
+        book = Book.objects.annotate(
+            authors_count=Count(
+                "authors",
+                filter=~Q(authors__name=[]),
+            ),
+        ).get(pk=self.b1.pk)
+        self.assertEqual(book.authors_count, 2)
+        aggregate = Book.objects.aggregate(
+            max_rating=Max("rating", filter=~Q(rating__in=[]))
+        )
+        self.assertEqual(aggregate, {"max_rating": 4.5})
```

Comment:
Two other cases that were not properly handling full matches. Also added tests for empty matches.